File: src/test/java/eu/sig/training/ch04/v3/CheckingAccountTest.java
package eu.sig.training.ch04.v3;

import org.junit.Before;
import org.junit.Test;

import eu.sig.training.ch04.BusinessException;
import eu.sig.training.ch04.Money;

public class CheckingAccountTest {

    CheckingAccount myAccount;
    CheckingAccount counterAccount;

    @Before
    public void setUp() {
        myAccount = Accounts.makeAccount(CheckingAccount.class, "123456789");
        counterAccount =
            Accounts.makeAccount(CheckingAccount.class, "497164833");
    }

    @Test(expected = BusinessException.class)
    public void TestTransferLimit() throws BusinessException {
        myAccount.makeTransfer(Accounts.getAccountNumber(counterAccount),
            new Money());
    }
}

File: src/test/csharp/eu/sig/training/ch04/v3/CheckingAccountTest.cs
using NUnit.Framework;

namespace eu.sig.training.ch04.v3
{

    [TestFixture]
    public class CheckingsAccountTest
    {

        CheckingAccount myAccount;
        CheckingAccount counterAccount;

        [SetUp]
        public void Init()
        {
            myAccount = Accounts.MakeAccount<CheckingAccount>("123456789");
            counterAccount = Accounts.MakeAccount<CheckingAccount>("497164833");
        }

        [Test]
        [ExpectedException(typeof(BusinessException))]
        public void TestTransferLimit()
        {
            myAccount.MakeTransfer(Accounts.GetAccountNumber(counterAccount), new Money());
        }
    }
}

